### PR TITLE
Eugene

### PR DESCRIPTION
**Pull request by NikolaevAS89**
1

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -79,3 +97,16 @@ services:
     ports:
       - 15672:15672
       - 5672:5672
+  ml:
```
Comment (NikolaevAS89) [inline docker-compose.yml]: Why is conteiner here?

File: audio-transcript/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+FROM python:3.10
 
 RUN mkdir "/application"
 RUN mkdir "/storage"
 COPY "requirements.txt" "/application/requirements.txt"
 COPY . "/application/"
 RUN ["pip", "install", "-r", "/application/requirements.txt"]
+RUN export LD_LIBRARY_PATH=`python3 -c 'import os; import nvidia.cublas.lib; import nvidia.cudnn.lib; print(os.path.dirname(nvidia.cublas.lib.__file__) + ":" + os.path.dirname(nvidia.cudnn.lib.__file__))'`
 WORKDIR "/application/scripts"
-ENTRYPOINT [ "python", "main.py"]
```
Comment (NikolaevAS89) [inline audio-transcript/Dockerfile:9]: why do you change ENTRYPOINT to CMD?

File: audio-transcript/scripts/main.py [hunk 3]
```diff
@@ -1,43 +1,90 @@
...
+connection = pika.BlockingConnection(pika.ConnectionParameters(host=rabbit_host,
                                                                credentials=credentials))
 
-queue_in_name = "audio.transcription.queue.in"
-queue_out_name = "audio.transcription.queue.out"
-exchange_name = "audio.transcription.exchange"
-routing_key_in = "audio.transcript.in"
-routing_key_out = "audio.transcript.out"
 
+class StorageService:
```
Comment (NikolaevAS89) [inline audio-transcript/scripts/main.py]: This class should be extracted to own file

File: audio-transcript/scripts/main.py [hunk 4]
```diff
@@ -1,43 +1,90 @@
...
-    answer = {
-        "uuid": bytes(body).decode("UTF-8"),
-        "words": [{
-            "word": "Свобода",
-            "start": "00:00:01",
-            "end": "00:00:03"
-        }]
-    }
-    #
+    storage = StorageService(path=grandparent_directory)
```
Comment (NikolaevAS89) [inline audio-transcript/scripts/main.py]: the variable storage should be global.

File: audio-transcript/scripts/main.py [hunk 5]
```diff
@@ -1,43 +1,90 @@
...
+    storage.get_source()
+    storage.split_source()
+    storage.generate_subs(uuid=uuid)
+
     ch.basic_publish(exchange=exchange_name,
                      routing_key=routing_key_out,
-                     body=json.dumps(answer).encode("UTF-8"))
-
+                     body=json.dumps(storage.answer).encode("UTF-8"))
+    print(storage.answer)
```
Comment (NikolaevAS89) [inline audio-transcript/scripts/main.py]: print should be changed to log

File: audio-transcript/scripts/main.py [hunk 6]
```diff
@@ -1,43 +1,90 @@
...
+    grandparent_directory = os.path.dirname(parent_directory)
+    if grandparent_directory not in sys.path:
+        sys.path.append(grandparent_directory)
+    uuid = body.decode("utf-8")
+
+    print(f" [x] Received   {str(body)}")
+    print(f" [x] Method     {str(method)}")
+    print(f" [x] Properties {str(properties)}")
+
     # TODO place code here.
```
Comment (NikolaevAS89) [inline audio-transcript/scripts/main.py]: delete this

File: audio-transcript/Dockerfile (hunk 2 above)
Comment (EugeneSeverin) [reply audio-transcript/Dockerfile:9]: because I'm tryyyying

**Review by EugeneSeverin — APPROVED** [review]
done

**Pull request merged**